shopfloor/services/zone_picking.py

```diff
@@ -637,12 +663,19 @@ def scan_source(self, barcode, confirmation=False, product_id=None):
         If a product is passed to the function the search on move line will
         be filtered based on it as well.
 
+        And if a sublocation_id is passed the search on move line will be restriced
+        to it.
+
         Transitions:
         * select_line: barcode not found or narrow the list on a location
         * set_line_destination: a line has been selected for picking
         """
-
         # select corresponding move line from barcode (location, package, product, lot)
+        sublocation = (
+            self.env["stock.location"].browse(sublocation_id)
```

Comment:
To protect against fake-ID
```suggestion
            self.env["stock.location"].browse(sublocation_id).exists()
```

Reply:
updated